openreview/arr/process/comment_process.py

```diff
@@ -0,0 +1,182 @@
+def process(client, edit, invitation):
+
+    domain = client.get_group(edit.domain)
+    venue_id = domain.id
+    meta_invitation_id = domain.get_content_value('meta_invitation_id')
+    short_name = domain.get_content_value('subtitle')
+    contact = domain.get_content_value('contact')
+    authors_name = domain.get_content_value('authors_name')
+    submission_name = domain.get_content_value('submission_name')
+    reviewers_name = domain.get_content_value('reviewers_name')
+    reviewers_anon_name = domain.get_content_value('reviewers_anon_name')
+    reviewers_submitted_name = domain.get_content_value('reviewers_submitted_name')
+    sender = domain.get_content_value('message_sender')
+    comment_threshold = domain.get_content_value('comment_notification_threshold')
+
+    submission = client.get_note(edit.note.forum, details='replies')
+    comment = client.get_note(edit.note.id)
+    paper_group_id=f'{venue_id}/{submission_name}{submission.number}'
+
+    ### TODO: Fix this, we should notify the use when the review is updated
+    if comment.tcdate != comment.tmdate:
+        return    
+
+    ignore_groups = comment.nonreaders if comment.nonreaders else []
+    ignore_groups.append(edit.tauthor)
+
+    signature = comment.signatures[0].split('/')[-1]
+    pretty_signature = openreview.tools.pretty_id(signature)
+    pretty_signature = 'An author' if pretty_signature == 'Authors' else pretty_signature
+
+    # Count comments between reviewer and authors
+    comment_count = 0
+    signed_by_reviewer = 'Reviewer' in signature
+    signed_by_author = 'Authors' in signature
+
+    for reply in submission.details['replies']:
+        if not reply['invitations'][0].endswith('Official_Comment'):
+            continue
+
+        readable_by_authors = any('Authors' in r for r in reply['readers'])
+        readable_by_reviewer = any('Reviewer' in r for r in reply['readers'])
+        readable_by_signature = any(comment.signatures[0] in r for r in reply['readers'])
+        if signed_by_reviewer:
+            readable_by_signature = readable_by_signature or any('Reviewers' in r for r in reply['readers'])
+
+        comment_by_author = any('Authors' in r for r in reply['signatures'])
+        comment_by_reviewer = any('Reviewer' in r for r in reply['signatures'])
+        comment_by_signature = comment.signatures[0] == reply['signatures'][0]
+
+        if not(readable_by_authors and readable_by_reviewer):
+            continue
+        ## Comments are readable by both authors and at least 1 reviewer
+        
+        # if new comment by reviewer, and (reply from author is readable by reviewer) or reply written by reviewer
+        if signed_by_reviewer and ((comment_by_author and readable_by_signature) or comment_by_signature):
```

Comment:
why does `signed_by_signature` and `comment_by_signature` count as a comment? What if it's not visible to the authors?

Reply:
There are some checks in the for loop that skip the reply if its not a comment, and checks that its readable by the Authors and some reviewer

Reply:
Oh actually in the new changes to count based on thread, I did make the filtering more explicit with these list comprehensions:
https://github.com/openreview/openreview-py/blob/ee86eb371219f035019f6409a60c7bd5b233f2c6/openreview/arr/process/comment_process.py#L48-L54